tests/test_update_password.py

```diff
@@ -0,0 +1,170 @@
+    faker, namespace, cleanup_handler, kopf_runner, api_client
+):
+    coapi = CustomObjectsApi(api_client)
+    core = CoreV1Api(api_client)
+    name = faker.domain_word()
+    password = faker.password(length=40)
+    new_password = faker.password(length=40)
+    username = faker.user_name()
+
+    cleanup_handler.append(
+        core.delete_persistent_volume(name=f"temp-pv-{namespace.metadata.name}-{name}")
+    )
+    await asyncio.gather(
+        core.create_namespaced_secret(
+            namespace=namespace.metadata.name,
+            body=V1Secret(
+                data={"password": b64encode(password)},
+                metadata=V1ObjectMeta(
+                    name=f"user-{name}", labels={LABEL_USER_PASSWORD: "true"}
+                ),
+                type="Opaque",
+            ),
+        ),
+    )
+
+    await coapi.create_namespaced_custom_object(
+        group=API_GROUP,
+        version="v1",
+        plural=RESOURCE_CRATEDB,
+        namespace=namespace.metadata.name,
+        body={
+            "apiVersion": "cloud.crate.io/v1",
+            "kind": "CrateDB",
+            "metadata": {"name": name},
+            "spec": {
+                "cluster": {
+                    "imageRegistry": "crate",
+                    "name": "my-crate-cluster",
+                    "version": "4.1.5",
+                },
+                "nodes": {
+                    "data": [
+                        {
+                            "name": "data",
+                            "replicas": 1,
+                            "resources": {
+                                "cpus": 0.5,
+                                "memory": "1Gi",
+                                "heapRatio": 0.25,
+                                "disk": {
+                                    "storageClass": "default",
+                                    "size": "16GiB",
+                                    "count": 1,
+                                },
+                            },
+                        }
+                    ]
+                },
+                "users": [
+                    {
+                        "name": username,
+                        "password": {
+                            "secretKeyRef": {
+                                "key": "password",
+                                "name": f"user-{name}",
+                            }
+                        },
+                    },
+                ],
+            },
+        },
+    )
+
+    host = await asyncio.wait_for(
+        get_public_host(core, namespace.metadata.name, name),
+        timeout=BACKOFF_TIME * 5,  # It takes a while to retrieve an external IP on AKS.
+    )
+
+    system_password = await get_system_user_password(
+        core, namespace.metadata.name, name
+    )
+    await assert_wait_for(
+        True,
+        is_password_set,
+        host,
+        system_password,
+        SYSTEM_USERNAME,
+        timeout=BACKOFF_TIME * 5,
+    )
```

Comment:
I don't think those two checks are needed. The `is_password_set` right below (line 148) will only succeed if a system user was created before.

Reply:
I just added them to make the test easier to understand ... but I can, of course, remove it.

Reply:
Na, it's fine :)